python_modules/dagster/dagster/_scheduler/scheduler.py

```diff
@@ -165,9 +168,19 @@ def launch_scheduled_runs(
         and schedule_state.status == InstigatorStatus.AUTOMATICALLY_RUNNING
     }
     for state in states_to_delete:
-        instance.schedule_storage.delete_instigator_state(
-            state.instigator_origin_id, state.selector_id
+        location_name = (
+            state.origin.external_repository_origin.repository_location_origin.location_name
         )
+        # don't clean up auto running state if its location is an error state
+        if location_name in error_locations:
+            break
+
+        if log_verbose_checks:
+            logger.info(
+                f"Removing state for automatically running schedule {state.instigator_name} "
+                f"that is no longer present in {location_name}"
+            )
```

Comment:
I don't think we need to wrap this one in log_verbose_checks since its relatively rare? The logging stuff is admittedly pretty funky but i think this was designed to guard against logging the same thing over and over again when nothing really happened, this falls under 'something happening' i think and i'd want it to always show up in the logs if we do it?